File: cs-core/11-extra/dsa-js-practice/sortingAlgos/insertionSort.js
function insertionSort(numbers) {
  for (let i = 0; i < numbers.length; i++) {
    const key = numbers[i];
    let j = i - 1;

    while (j >= 0 && numbers[j] > key) {
      numbers[j + 1] = numbers[j];
      j--;
    }

    numbers[j + 1] = key;
  }

  return numbers;
}

const coolNums = [1, 4, 5, 6, 3333, 555, 222, 34, 4, 5, 5, 6];

console.log(coolNums);
console.log();
console.log(insertionSort(coolNums));

File: cs-core/02-programming-fundamentals/04-sorting-and-searching/insertionSort.cpp
#include <iostream>

using std::cout, std::endl;

void insertionSort(int *numbers, int size)
{
  for (int i = 1; i < size; i++)
  {
    int current = numbers[i];
    int j;

    for (j = i - 1; j >= 0 && numbers[j] > current; j--)
    {
      numbers[j + 1] = numbers[j];
    }

    numbers[j + 1] = current;
  }
}

int main()
{
  int numbers[] = {3, 5, 6, 7, 8, 9, 222, 34454, 111, 3};
  int size = sizeof(numbers) / sizeof(numbers[0]);

  cout << "Original array: ";
  for (int i = 0; i < size; i++)
    cout << numbers[i] << " ";
  cout << endl;

  insertionSort(numbers, size);

  cout << "Sorted array: ";
  for (int i = 0; i < size; i++)
    cout << numbers[i] << " ";
  cout << endl;

  return 0;
}